sycl/include/sycl/ext/intel/experimental/esimd/memory.hpp

```diff
@@ -3246,11 +3251,13 @@ template <__ESIMD_NS::atomic_op Op, typename T, int N,
           lsc_data_size DS = lsc_data_size::default_size,
           cache_hint L1H = cache_hint::none, cache_hint L3H = cache_hint::none,
           typename AccessorTy>
-__ESIMD_API std::enable_if_t<
-    sycl::detail::acc_properties::is_local_accessor_v<AccessorTy>,
-    __ESIMD_NS::simd<T, N>>
-lsc_atomic_update(AccessorTy acc, __ESIMD_NS::simd<uint32_t, N> offsets,
-                  __ESIMD_NS::simd<T, N> src0, __ESIMD_NS::simd_mask<N> pred) {
+__ESIMD_API
+    std::enable_if_t<__ESIMD_DNS::is_local_accessor_with_v<
+                         AccessorTy, __ESIMD_DNS::accessor_mode_cap::can_write>,
```

Comment:
I am not sure this is correct. The atomic_operation (e.g. fmin) is some sort of 3-in-1: read+fmin+store.

The access permissions for accessors could potentially be used by scheduler to know how to sync memory on host and device. 
Read means the buffer accessed by accessor must be copied from HOST to DEVICE (while write-only access means that need only allocate memory, without copying actual data from HOST).

Reply:
Right, that is what I mean. The accessor needs the check not only for read permission, but for write as well.
I.e. either has accessor_mode_cap::write && has accessor_mode_cap::read
or has accessor_mode_cap::read_write. 2nd would require adding read_write enum value.